src/EventStore.ClientAPI/EventStoreConnection.cs

```diff
@@ -107,9 +107,16 @@ public static IEventStoreConnection
 				}
 
 				if (scheme == "discover") {
-					var clusterSettings = new ClusterSettings(uri.Host, connectionSettings.MaxDiscoverAttempts,
-						uri.Port,
+					var port = uri.Port == -1 ? 2113 : uri.Port;
```

Comment:
It seems that the default port should be set to `connectionSettings.GossipPort` here to maintain backwards compatibility.

Reply:
Done, tell me if it's better now.